windows/devices/uwb/UwbSession.cxx

```diff
@@ -178,69 +178,40 @@ UwbStatus
-    // // TODO: all other memory required for this structure must be accounted for, the above calculation was left incomplete.
-    // // Also, proper memory alignment of trailing structures in the allocated buffer has not been taken into account.
-    // auto appConfigParamsBuffer = std::make_unique<uint8_t[]>(appConfigParamsSize);
-    // auto *appConfigParams = reinterpret_cast<UWB_SET_APP_CONFIG_PARAMS *>(appConfigParamsBuffer.get());
-    // appConfigParams->sessionId = GetId();
-    // appConfigParams->appConfigParamsCount = 2;
-    // UWB_APP_CONFIG_PARAM *appConfigParamList = reinterpret_cast<UWB_APP_CONFIG_PARAM *>(appConfigParams + 1);
-    // UWB_APP_CONFIG_PARAM *appConfigParamNumberOfControlees = appConfigParamList + 0;
-    // UWB_APP_CONFIG_PARAM *appConfigParamDstMacAddress = appConfigParamList + 1;
-
-    // // Populate NUMBER_OF_CONTROLEES app configuration parameter.
-    // auto &numberOfControleesPayload = *reinterpret_cast<uint8_t *>(appConfigParamNumberOfControlees + 1);
-    // appConfigParamNumberOfControlees->paramType = UWB_APP_CONFIG_PARAM_TYPE_NUMBER_OF_CONTROLEES;
-    // appConfigParamNumberOfControlees->paramLength = 1;
-    // numberOfControleesPayload = static_cast<uint8_t>(m_peers.size());
-
-    // // Populate DST_MAC_ADDRESS app configuration parameter.
-    // auto dstMacAddressPayload = reinterpret_cast<uint8_t *>(appConfigParamDstMacAddress + 1);
-    // appConfigParamDstMacAddress->paramType = UWB_APP_CONFIG_PARAM_TYPE_DST_MAC_ADDRESS;
-    // appConfigParamDstMacAddress->paramLength = static_cast<uint32_t>(macAddressessLength);
-    // auto dstMacAddress = dstMacAddressPayload;
-    // for (const auto &peer : m_peers) {
-    //     const auto value = peer.GetValue();
-    //     std::copy(std::cbegin(value), std::cend(value), dstMacAddress);
-    //     std::advance(dstMacAddress, std::size(value));
-    // }
-
-    // // Attempt to set all new parameters.
-    // DWORD bytesReturned = 0;
-    // UWB_SET_APP_CONFIG_PARAMS_STATUS appConfigParamsStatus; // TODO: this needs to be dynamically allocated to fit returned content
-    // BOOL ioResult = DeviceIoControl(m_handleDriver.get(), IOCTL_UWB_SET_APP_CONFIG_PARAMS, &appConfigParams, static_cast<DWORD>(appConfigParamsSize), &appConfigParamsStatus, sizeof appConfigParamsStatus, &bytesReturned, nullptr);
-    // if (!LOG_IF_WIN32_BOOL_FALSE(ioResult)) {
-    //     // TODO
-    //     HRESULT hr = GetLastError();
-    //     PLOG_ERROR << "could not send params to driver, hr=" << std::showbase << std::hex << hr;
-    // }
+    if (not m_uwbSessionConnector) {
+        PLOG_WARNING << "No associated connector";
+        return UwbStatusGeneric::Failed;
+    }
+
+    auto params = GetApplicationConfigurationParameters({ UwbApplicationConfigurationParameterType::DestinationMacAddresses });
+    if (std::size(params) != 1) {
+        PLOG_ERROR << "did not return the correct number of parameters";
+        return UwbStatusGeneric::Failed;
```

Comment:
This is a programmer error and we should flag it immediately. Such code should never get into the wild; ideally, we'll develop a test to fuzz the API and ensure this never happens, but if it does, this exception will flag it so we can fix it:

```suggestion
                throw std::runtime_error("GetApplicationConfigurationParameters() for 1 parameter returned > 1 results. This is a bug!");
```